src/components/Filter/components/predefinedFilter.js

```diff
@@ -1,33 +1,56 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect, useCallback } from 'react';
 
 type OptionShape = {
   text: string,
   value: string,
 };
 
 type PredefinedFilterProps = {
-  dropdownRef: any,
-  onToggle: Function,
   options: OptionShape[],
   onSelect: Function,
   baseClass?: string,
   filterButtonContent?: Function,
 };
 
 const PredefinedFilter = ({
-  dropdownRef,
-  onToggle,
   options,
   onSelect,
   baseClass,
   filterButtonContent,
 }: PredefinedFilterProps) => {
   const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef(null);
+
+  const listenerAction = useCallback(() => {
+    setIsOpen(false);
```

Comment:
note that we'll also need these lines-

https://github.com/bonlineza/b-lib/pull/70/files#diff-b1a557af2754e767b3b2633f6be4eb94L193-L204

otherwise if you click in the popup the popup will close

Reply:
done